crates/typst/src/foundations/float.rs

```diff
@@ -47,6 +47,21 @@ impl f64 {
     ) -> f64 {
         value.0
     }
+
+    /// Checks if a float is not a number.
+    ///
+    /// In IEEE 754, not all bit patterns are valid floats. This function
+    /// returns `true` if the float is not a number (NaN).
+    ///
+    /// ```example
+    /// #float.is-nan(0) \
+    /// #float.is-nan(1) \
+    /// #float.is-nan(calc.nan)
+    /// ```
+    #[func]
+    pub fn is_nan(self) -> bool {
+        self.is_nan()
```

Comment:
How does this work? `f64` type is redefined, so how is it possible to call `f64::is_nan`? Is it because of macro?

Reply:
It works using the macros

Reply:
The `type f64` is erased by the macro, it's just the to communicate the type to the macro. For primitive types like this the `#[scope]` macro generates an extension trait under the hood to put the functions into. Since is_nan is defined as an inherent function on f64, `self.is_nan()` dispatches to that one instead of the one new one in the extension trait. It took me a bit to figure that out what's happening though, so perhaps it could be made clearer that this isn't infinitely recursive by writing `f64::is_nan(self)` instead.